### PR TITLE
GA Migration PostCommit Java Examples Direct, Flink, Spark, Hadoop Versions

### PR DESCRIPTION
**Pull request by fernando-wizeline**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

Migrated workflow [job-postcommit-java-examples-direct.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/.github/workflows/job-postcommit-java-examples-direct.yml)
Migrated workflow [job-postcommit-java-examples-flink.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/.github/workflows/job-postcommit-java-examples-flink.yml)
Migrated workflow [job-postcommit-java-examples-spark.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/.github/workflows/job-postcommit-java-examples-spark.yml)
Migrated workflow [job-postcommit-java-hadoop-versions.yml](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/.github/workflows/job-postcommit-java-hadoop-versions.yml)

The migrated workflow was removed from [.test-infra/jenkins/README.md](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/.test-infra/jenkins/README.md) and added to [CI.md](https://github.com/fernando-wizeline/beam/blob/gam-postcommit-java-examples-direct-flink-spark/CI.md)

DO NOT MERGE until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-java-examples-spark.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    name: Run PostCommit Java Examples Spark
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix: #JDK17 execution will fail due to new access policies applied to the core JDK libraries starting at JDK 16.
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        spark: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SPARK_VERSIONS}}
+        exclude:
+          - java: "11"
+            spark: "2"
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-examples-spark.yml:58]: Is this excluded for the same reason as 17? If not, please comment to explain

File: .github/workflows/job-postcommit-java-hadoop-versions.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    needs: set-properties
+    name: Run PostCommit Java Hadoop Versions
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        exclude:
+          - java: "11"
+          - java: "17"
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-hadoop-versions.yml:57]: Please add a comment explaining why these are excluded + linking to the issue

File: .github/workflows/job-postcommit-java-examples-spark.yml (hunk 1 above)
Comment (damccorm) [reply .github/workflows/job-postcommit-java-examples-spark.yml:58]: Sorry to be pedantic, but could you please add some more detail here? Compatibility issues doesn't really tell me much about what the problem is

Comment (fernando-wizeline) [reply .github/workflows/job-postcommit-java-examples-spark.yml:58]: > Sorry to be pedantic, but could you please add some more detail here? Compatibility issues doesn't really tell me much about what the problem is

Oh, no worries, Danny!
About the compatibility issues, Spark v2 dropped support for JDK9+. Spark v3 does work with JDK9+. This thread goes into further detail: https://issues.apache.org/jira/browse/SPARK-24421

Should I add the comment mentioning that?
Thanks!

Comment (damccorm) [reply .github/workflows/job-postcommit-java-examples-spark.yml:58]: Yep - just something like `Spark v2 dropped support for JDK9+` is great, thanks!

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**